tools/check_doc.py

```diff
@@ -4,98 +4,173 @@
-    valid = True
-    code_blocks = []
-    state = State.OUTSIDE
-
-    # reStructuredText is not yet supported
-    # https://github.com/Componolit/RecordFlux/issues/880
-    for p in Path("doc").rglob("*.rst"):
-        is_markdown = p.suffix == ".md"
-        if is_markdown:
-            prefix = "```"
-            separator = "```"
-        else:
-            prefix = "[source,"
-            separator = "----"
-        for l in p.read_text():
-            if state == State.OUTSIDE and l.startswith(prefix):
-                if is_markdown:
-                    state = State.INSIDE
-                else:
-                    state = State.HEADER
-                block = ""
-                if l.startswith(f"{prefix}ada,rflx"):
-                    block_type = CodeBlockType.RFLX
-                    subtype = l[len(prefix) + 9 : -2]
-                elif l.startswith(f"{prefix}ada"):
-                    block_type = CodeBlockType.ADA
-                elif l.startswith(f"{prefix}python"):
-                    block_type = CodeBlockType.PYTHON
-                else:
-                    block_type = CodeBlockType.UNKNOWN
-                continue
-
-            if state == State.HEADER and l == f"{separator}\n":
-                state = State.INSIDE
-                continue
-
-            if state == State.INSIDE and l == f"{separator}\n":
-                state = State.OUTSIDE
-                code_blocks.append((block_type, subtype, block))
-                continue
-
-            if state == State.INSIDE:
-                block += l
-
-    ("build" / SPEC_DIR.parent).mkdir(parents=True, exist_ok=True)
-    os.symlink(os.getcwd() / SPEC_DIR, "build" / SPEC_DIR)
-    os.chdir("build")
-
-    for block_type, subtype, block in code_blocks:
-        valid = check_code(block, block_type, subtype) and valid
-
-    os.unlink(SPEC_DIR)
+def parse_code_block_type(type_str: str) -> CodeBlockType:
+    normalized = type_str.lower()
+    types = {
+        "rflx": CodeBlockType.RFLX,
+        "ada": CodeBlockType.ADA,
+        "python": CodeBlockType.PYTHON,
+        "yaml": CodeBlockType.YAML,
+        "ignore": CodeBlockType.IGNORE,
+    }
+    if normalized not in types:
+        return CodeBlockType.UNKNOWN
+    return types[normalized]
 
-    return valid
 
+def check_file(filename: Path, content: str) -> None:
+    # pylint: disable=too-many-branches, too-many-statements
 
-def check_code(block: str, block_type: CodeBlockType, subtype: str = None) -> bool:
-    if block_type == CodeBlockType.RFLX:
-        assert subtype is not None
-        return check_rflx_code(block, subtype)
-    if block_type is CodeBlockType.ADA:
-        return check_ada_code(block)
-    if block_type is CodeBlockType.PYTHON:
-        return check_python_code(block)
-    return True
+    found = False
+    state = State.OUTSIDE
+    block = ""
+    block_start: Optional[int] = None
+    doc_type: Optional[CodeBlockType] = None
```

Comment:
The name `doc_type` confused me. How about renaming `doc_type` to `doc_check_type` and `block_type` to `code_type` to make them more similar to the directive type names in the RST file?

Reply:
Changed.